packages/dx-react-grid/src/components/table-layout.jsx

```diff
@@ -19,19 +19,37 @@ export class TableLayout extends React.PureComponent {
     };
 
     this.animations = new Map();
+    this.savedScrolldWidth = {};
   }
 
   componentDidUpdate(prevProps) {
     const { columns } = this.props;
     const { columns: prevColumns } = prevProps;
 
-    // eslint-disable-next-line react/no-find-dom-node
-    const tableWidth = findDOMNode(this).scrollWidth;
+    const tableWidth = this.getTableWidth(prevColumns, columns);
     this.animations = getAnimations(prevColumns, columns, tableWidth, this.animations);
+
     cancelAnimationFrame(this.raf);
     this.raf = requestAnimationFrame(this.processAnimationFrame.bind(this));
   }
 
+  getTableWidth(prevColumns, columns) {
+    // eslint-disable-next-line react/no-find-dom-node
+    const { offsetWidth, scrollWidth } = findDOMNode(this);
```

Comment:
Can we avoid to use a `findDOMNode` function? Because `strictMode` said that this is a deprecated API.

Reply:
Yes, we can try to do it since tableLayout is not public